common/djangoapps/course_about/tests/test_views.py

```diff
@@ -0,0 +1,164 @@
+"""
+Tests for user enrollment.
+"""
+import ddt
+import json
+import unittest
+
+from django.test.utils import override_settings
+from django.core.urlresolvers import reverse
+from rest_framework.test import APITestCase
+from rest_framework import status
+from django.conf import settings
+from xmodule.modulestore.tests.django_utils import (
+    ModuleStoreTestCase, mixed_store_config
+)
+from xmodule.modulestore.tests.factories import CourseFactory, CourseAboutFactory
+from student.tests.factories import UserFactory
+from util.parsing_utils import course_image_url
+from course_about import api
+from course_about.errors import CourseNotFoundError, CourseAboutError
+from mock import patch
+from xmodule.modulestore.django import modulestore
+from datetime import datetime
+
+# Since we don't need any XML course fixtures, use a modulestore configuration
+# that disables the XML modulestore.
+
+MODULESTORE_CONFIG = mixed_store_config(settings.COMMON_TEST_DATA_ROOT, {}, include_xml=False)
+
+
+@ddt.ddt
+@override_settings(MODULESTORE=MODULESTORE_CONFIG)
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class CourseInfoTest(ModuleStoreTestCase, APITestCase):
+    """
+    Test course information.
+    """
+    USERNAME = "Bob"
+    EMAIL = "bob@example.com"
+    PASSWORD = "edx"
+
+    def setUp(self):
+        """ Create a course"""
+        super(CourseInfoTest, self).setUp()
+
+        self.course = CourseFactory.create()
+        self.user = UserFactory.create(username=self.USERNAME, email=self.EMAIL, password=self.PASSWORD)
+        self.client.login(username=self.USERNAME, password=self.PASSWORD)
+
+    def test_user_not_authenticated(self):
+        # Log out, so we're no longer authenticated
+        self.client.logout()
+
+        resp = self.client.get(
+            reverse('courseabout', kwargs={"course_id": unicode(self.course.id)})
+        )
+        resp_data = json.loads(resp.content)
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertIsNotNone(resp_data)
+
+    def test_with_valid_course_id(self):
+        resp = self.client.get(
+            reverse('courseabout', kwargs={"course_id": unicode(self.course.id)})
+        )
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+
+    def test_with_invalid_course_id(self):
+        resp = self.client.get(
+            reverse('courseabout', kwargs={"course_id": 'not/a/validkey'})
+        )
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_get_course_details_all_attributes(self):
+        kwargs = dict()
+        kwargs["course_id"] = self.course.id
+        kwargs["course_runtime"] = self.course.runtime
+        CourseAboutFactory.create(**kwargs)
+        resp = self.client.get(
+            reverse('courseabout', kwargs={"course_id": unicode(self.course.id)})
+        )
+        resp_data = json.loads(resp.content)
```

Comment:
These lines seem to be re-used in a lot of tests.  Please move them into a helper function, something like `resp_data = _get_course_about(self.course_id)`